signer/xpi/jar.go

```diff
@@ -302,6 +306,20 @@ func isJARSignatureFile(name string) bool {
 	return false
 }
 
+// The XPI format reserves a couple number signature files stored under the META-INF directory
+// META-INF/COSE.SIG
+// META-INF/COSE.MANIFEST
+// and their lower and mixed case variants
+func isCOSESignatureFile(name string) bool {
+	if strings.HasPrefix(name, "META-INF/") {
+		name = strings.ToLower(strings.TrimPrefix(name, "META-INF/"))
+		if name == "cose.manifest" || name == "cose.sig" {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
iiuc, this new function and the change on line 220 is all that's needed to address the bug. everything else is testing, right?

Reply:
Yep! This and the changes on lines 127 and 220 are the only code changes.

Everything else is test / client. I didn't know what was going on with the AMO issue, so I kept writing tests for invariants until it failed.

Improvements (TODO later) 1) check each specific meta file and validate its format (instead of just counting them), 2) check generated manifests verify, and 3) fuzz it all w/ libfuzzer go-fuzz